### PR TITLE
Solves kata 8 - PATCH /api/articles/:article_id

### PR DESCRIPTION
**Pull request by samiracaruso**
Specifically:
-Implements logic to update the article votes and returns the updated article;
-Handles server errors (500), article not found (404), incorrect article id type (400) and incorrect votes type (400)

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -205,4 +205,57 @@ test('Responds with an error 401 for an invalid username', () => {
...
+    test('Responds with a 400 if the inc_votes data type is invalid', () => {
+        const votes = {inc_votes: 'one'}
+        return request(app)
+        .patch('/api/articles/3')
+        .send(votes)
+        .expect(400)
+        .then(({body}) => {
+            expect(body.error.message).toBe('Bad Request: Invalid inc_votes data')
+        })
+    })
```
Comment (nc-rayhaan) [inline __tests__/app.test.js:260]: Brilliant tests! 👏 1 more test would be helpful too, what if the inc_votes object is empty?

File: app/controllers/controller.js [hunk 2]
```diff
@@ -71,4 +77,20 @@ exports.addComment = (req, res, next) => {
...
     })
     .catch(next)
+}
+
+exports.updateArticleById = (req, res, next) => {
+    const {article_id} = req.params
+    const {inc_votes} = req.body
+    if (isNaN(article_id) || !Number.isInteger(Number(article_id)) || Number(article_id) <= 0) {
+        return next({ status: 400, message: 'Bad Request: Invalid article id' });
+    }
```
Comment (nc-rayhaan) [inline app/controllers/controller.js:87]: This logic should ideally be in the model with a promise reject.

**Review by nc-rayhaan — COMMENTED** [review]
Great work! 😁
